### PR TITLE
chore(mme): Delete dead code in mem_app_sgsap.c

### LINKED ISSUE #11574: Delete duplicate code in mme_app_sgsap.c
Discovered during Bazel build debug, this c file provides the function
mme_app_handle_sgsap_paging_request, which is also provided in
mme_app_sgs_paging.c.

### PR DESCRIPTION
**Pull request by electronjoe**
Discovered during Bazel build debug, this c file provides the function
mme_app_handle_sgsap_paging_request, which is also provided in
mme_app_sgs_paging.c.

I looked to see if there were any header modifications necessary but we have only the one header exporting the function.

Closes #11574.

## Test Plan

CI tests which include `make build_oai` and `make test_oai`.

Signed-off-by: Scott Moeller <electronjoe@gmail.com>

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/mme_app/CMakeLists.txt [hunk 1]
```diff
@@ -36,11 +36,10 @@ add_library(TASK_MME_APP
     mme_app_sgs_null.c
     mme_app_sgs_la_updated.c
     mme_app_sgs_associated.c
-    mme_app_sgs_reset.c
+    mme_app_sgs_reset.cmme_app_sgsap.c
```
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/mme_app/CMakeLists.txt]: File mme_app_sgsap.c is deleted, shall we remove from CMakeLists.txt file as well

Comment (electronjoe) [reply lte/gateway/c/core/oai/tasks/mme_app/CMakeLists.txt]: I'm not sure what happened here, thanks for the catch! I presume this doesn't compile right now (going to check CI results for `make build_oai`).

Comment (electronjoe) [reply lte/gateway/c/core/oai/tasks/mme_app/CMakeLists.txt]: Indeed the CI pipeline detected my error here (thankfully). Thank you for the review! I've fixed up the PR and pushed a revision.

**Review by ajahl — APPROVED** [review]

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**